sycl/test-e2e/lit.cfg.py

```diff
@@ -36,18 +39,33 @@
 # To be filled by lit.local.cfg files.
 config.required_features = []
 config.unsupported_features = []
+# Get the set of all features possible in E2E tests.
+all_features = get_all_sycl_lit_features()
+
+# Please do not call config.available_features.add directly. Use this function.
+# This wrapper function checks if the feature is contained in all_features
+# and updates available_features. It ensures all new available features are added
+# to all_features.
+def register_feature(config, all_features, feature_name, condition):
+    if feature_name not in all_features:
+        raise Exception(
+            feature_name
+            + " is not presented in all_features. Please update the set in sycl/test-e2e/sycl_lit_features.py."
+        )
+    if condition:
+        config.available_features.add(feature_name)
 
 # test-mode: Set if tests should run normally or only build/run
 config.test_mode = lit_config.params.get("test-mode", "full")
 config.fallback_build_run_only = False
 if config.test_mode == "full":
-    config.available_features.add("run-mode")
-    config.available_features.add("build-and-run-mode")
+    register_feature(config, all_features, "run-mode", True)
```

Comment:
Can we change the type of `config.available_features` so that its `.add()` would do the right thing automatically?

Reply:
You can just add `.add()` method to whatever type is returned by `get_all_sycl_lit_features()`, rename that variable into `available_features` to make the code look closer to its current state.

Also: as long as our thing mimics Python array interface, we can just completely replace `config.available_features` with our own thing, i.e. `config.available_features = our_custom_features_registry()`

Reply:
@aelovikov-intel I guess that's what you're looking for - [0976851](https://github.com/intel/llvm/pull/16019/commits/097685122cf967462e0933ccd1ae90d87263001e)